samples/string_interning.cpp

```diff
@@ -0,0 +1,18 @@
+#include <vector>
+#include <string>
+
+char* add_string(char *string, std::vector<char*> &old_strings) {
+    //check if the string is stored
+    for (size_t i = 0; i < old_strings.size(); i++) {
+        if (!strcmp(string, old_strings[i])) {
```

Comment:
One good way to optimize this a little further is to save the length of each string apart from the string itself. That way, you can start comparing the strings only if their lenghts match. Because most of the time their lenghts won't match and so you'll just go to the next.